pkgs/tools/networking/rathole/default.nix

```
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+, Security
+, CoreServices
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "rathole";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "rapiz1";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-gqWgx03mUk6+9K4Yw5PHEBwFxsOR+48wvngT+wQnN1k=";
+  };
+
+  cargoSha256 = "sha256-dafOgZtiszkoi97PpAVMtdvJd5O3EK9hDVNLJ32FYzE=";
+  buildInputs = [ openssl ] ++ lib.optional stdenv.isDarwin [Security CoreServices];
+
+  nativeBuildInputs = [ pkg-config ];
```

```suggestion

  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ CoreServices ];

  __darwinAllowLocalNetworking = true;
```

A better order is `nativeBuildInputs` and then `buildInputs`.

`__darwinAllowLocalNetworking` is needed to build with darwin sandbox enabled.